sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -247,6 +251,67 @@ object Literal {
       s"Literal must have a corresponding value to ${dataType.catalogString}, " +
       s"but class ${Utils.getSimpleName(value.getClass)} found.")
   }
+
+  /**
+   * Parse and analyze the .sql string of a Literal, construct the Literal given the data type json.
+   * Example usage:
+   *   val serializedValue = lit.sql
+   *   val dataTypeJson = lit.dataType.json
+   *   val deserializedLit: Literal = Literal.fromSQL(serializedValue, dataTypeJson)
+   *
+   * @param valueSqlStr the .sql string of the Literal
+   * @param dataTypeJson the json format data type of the Literal
+   * @throws AnalysisException INVALID_LITERAL_VALUE_SQL_STRING_FOR_DESERIALIZATION
+   */
+  private[sql] def fromSQL(valueSqlStr: String, dataTypeJson: String): Literal = {
+    def parseAndAnalyze(valueSqlStr: String, dataType: DataType): Expression = {
```

Comment:
This method is very similar to https://github.com/apache/spark/blob/e6b4fa835de3f6d0057bf3809ea369d785967bcd/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala#L263

Could we reuse the `ResolveDefaultColumns.analyze`?

Reply:
I was thinking about that too but I found that `ResolveDefaultColumns.analyze` has 1) more column default specific error and error messages 2) has more limitation on the supported case. I also found it can return a non-Literal value as return, e.g. a Cast. 
Instead I think in the future ResolveDefaultColumns could refactor to use this general helper function.